types/src/proof/definition.rs

```diff
@@ -237,6 +237,98 @@ impl SparseMerkleProof {
     pub fn siblings(&self) -> &[HashValue] {
         &self.siblings
     }
+
+    /// If `element_blob` is present, verifies an element whose key is `element_key` and value is
+    /// `element_blob` exists in the Sparse Merkle Tree using the provided proof. Otherwise
+    /// verifies the proof is a valid non-inclusion proof that shows this key doesn't exist in the
+    /// tree.
+    pub fn verify(
+        &self,
+        expected_root_hash: HashValue,
+        element_key: HashValue,
+        element_blob: &Option<AccountStateBlob>,
```

Comment:
This mirrors the original, but since we are now here, don't you think this should be `Option<&AccountStateBlob>` instead?  I don't think a reference to an option makes very much sense.

Reply:
Oh I didn't pay attention to this. Yeah, I think `Option<&>` makes sense, not sure why it's `&Option`.

Reply:
Internally, Rust converts `Option<&T>` into the equivalent of nullptr for the `None` case, but the optimization is disabled if the type is the other way around and you have to deal with extra reference semantics on the outer type.  Apparently `&Option<T>` is a clever way to get around some `FnOnce` semantics, but that already seems unsound - `FnOnce` is built that way for a reason.